src/ol/View.js

```diff
@@ -184,46 +196,62 @@ const DEFAULT_MIN_ZOOM = 0;
  *
  * `setCenter`, `setResolution` and `setRotation` can be used to change the
- * states of the view. Any value can be passed to the setters. And the value
- * that is passed to a setter will effectively be the value set in the view,
- * and returned by the corresponding getter.
- *
- * But a View object also has a *resolution constraint*, a
- * *rotation constraint* and a *center constraint*.
- *
- * As said above, no constraints are applied when the setters are used to set
- * new states for the view. Applying constraints is done explicitly through
- * the use of the `constrain*` functions (`constrainResolution` and
- * `constrainRotation` and `constrainCenter`).
+ * states of the view, but any constraint defined in the constructor will
+ * be applied along the way.
  *
- * The main users of the constraints are the interactions and the
- * controls. For example, double-clicking on the map changes the view to
- * the "next" resolution. And releasing the fingers after pinch-zooming
- * snaps to the closest resolution (with an animation).
+ * A View object can have a *resolution constraint*, a *rotation constraint*
+ * and a *center constraint*.
  *
- * The *resolution constraint* snaps to specific resolutions. It is
- * determined by the following options: `resolutions`, `maxResolution`,
- * `maxZoom`, and `zoomFactor`. If `resolutions` is set, the other three
- * options are ignored. See documentation for each option for more
- * information.
+ * The *resolution constraint* typically restricts min/max values and
+ * snaps to specific resolutions. It is determined by the following
+ * options: `resolutions`, `maxResolution`, `maxZoom`, and `zoomFactor`.
+ * If `resolutions` is set, the other three options are ignored. See
+ * documentation for each option for more information. By default, the view
+ * only has a min/max restriction and allow intermediary zoom levels when
+ * pinch-zooming for example.
  *
  * The *rotation constraint* snaps to specific angles. It is determined
  * by the following options: `enableRotation` and `constrainRotation`.
  * By default the rotation value is snapped to zero when approaching the
  * horizontal.
  *
  * The *center constraint* is determined by the `extent` option. By
- * default the center is not constrained at all.
+ * default the view center is not constrained at all.
+ *
+ * ### Changing the view state
+ *
+ * Using only zoom, resolution, center and rotation setters to manipulate the
+ * view can cause issues because the way constraints are applied implicitly.
+ * Typically, calling `setResolution(10)` does not guarantee that
+ * `getResolution()` will return `10`.
  *
-  * @api
+ * To avoid this, one can use `move`, `adjustRotation`, `adjustZoom` and
+ * `adjustResolution` in order to change the view state while letting the
+ * view do its internal computations. The `adjust*` methods also take an `opt_anchor`
+ * argument which allows specifying an origin for the transformation.
+ *
```

Comment:
Does this mean that using the `adjust*` methods allows users to bypass constraints? What will be the behavior of the view when a constraint is broken this way?

Reply:
They do not bypass constraints but simply apply the given delta to the corresponding `target*` value, handle the computation related to the anchor and eventually call the associated `set*` methods (which goes through the constraint system).

With this refactoring an external consumer should not do these computations by itself (ie get the current value, apply a delta, take into account an anchor, and call `set*` with the result) as the result might be incorrect because of contraints.